### PR TITLE
LINK-1924 | Adds possibility to manage organization accounts

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Adds the possibility to add and edit Talpa accounts in organization edit page. Deleting is not possible as it is not possible in Talpa either. Only superusers and financial admins are allowed to edit accounts and the account related fields are disabled for all the other users.

## Closes
[LINK-1924](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1924)

## Screenshot
Screenshot 2024-04-08 at 11 36 32

[LINK-1924]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1924?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/app/i18n/fi.json [hunk 1]
```diff
@@ -1619,8 +1619,21 @@
       "title": "Varmista organisaation poistaminen"
     },
     "form": {
+      "account": {
+        "labelActive": "Aktiivinen",
+        "labelBalanceProfitCenter": "Tasapainotettu tulosyksikkö",
+        "labelCompanyCode": "SAP yrityskoodi",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/fi.json:1625]:
```suggestion
        "labelCompanyCode": "SAP-yrityskoodi",
```

File: src/domain/app/i18n/sv.json [hunk 2]
```diff
@@ -1619,8 +1619,21 @@
       "title": "Bekräfta raderingen av organisationen"
     },
     "form": {
+      "account": {
+        "labelActive": "Är aktiv",
+        "labelBalanceProfitCenter": "Balansera resultatenheten",
+        "labelCompanyCode": "SAP företagskod",
+        "labelInternalOrder": "Intern ordning",
+        "labelMainLedgerAccount": "Huvudbokkonto",
+        "labelOperationArea": "SAP:s funktionsområde",
```
Comment (harriris-vincit) [inline src/domain/app/i18n/sv.json:1628]:
```suggestion
        "labelOperationArea": "SAP funktionsområde",
```

File: src/domain/app/i18n/fi.json (hunk 1 above)
Comment (jorilindell) [reply src/domain/app/i18n/fi.json]: Changed as suggested

File: src/domain/organization/hooks/useOrganizationActions.ts [hunk 3]
```diff
@@ -139,30 +148,75 @@ const useOrganizationUpdateActions = ({
...
         savingFinished,
       });
     }
   };
 
+  const updateOrganization = async (
+    values: OrganizationFormFields,
+    callbacks?: MutationCallbacks<string>
+  ) => {
+    if (
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/organization/hooks/useOrganizationActions.ts]: Changed as suggested

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr362-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr362-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr362-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr362-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/organization/organizationForm/__tests__/OrganizationForm.test.tsx [hunk 4]
```diff
@@ -28,19 +28,52 @@ const renderComponent = () =>
...
     case 'paytrailMerchantIdInput':
       return screen.getByRole('textbox', { name: 'Paytrail-kauppiastunnus *' });
+    case 'vatCodeInput':
+      return screen.getByRole('textbox', { name: 'ALV-koodi *' });
   }
 };
 
-test('should add and remove price group', async () => {
+test('should add and remove account', async () => {
+  mockAuthenticatedLoginState();
```
Comment (harriris-vincit) [inline src/domain/organization/organizationForm/__tests__/OrganizationForm.test.tsx:56]: Optional suggestion (can merge without this in my opinion): SonarCloud seems to have picked up duplication, for example in these tests. Maybe the tests could benefit from scoping with `describe` or from a `beforeEach` to have the initialization steps repeated only once?

**Review by harriris-vincit — APPROVED** [review]
Looks good IMHO. One optional suggestion in the comments about reducing code duplication.

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr362-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**